pkg/distributors/contracts/MerkleOrchard.sol

```diff
@@ -63,16 +63,57 @@ contract MerkleOrchard {
     ) internal {
         uint256[] memory amounts = new uint256[](tokens.length);
 
+        // To save gas when setting claimed statuses in storange we group updates
+        // into currentBits for a particular channel, only setting them when a claim
+        // on a new channel is seen, or on the final iteration
+
+        // for aggregating claims
+        uint256 currentBits;
+        bytes32 currentChannelId;
+        uint256 currentWordIndex;
+        uint256 currentClaimAmount;
+
         Claim memory claim;
-        IERC20 token;
         for (uint256 i = 0; i < claims.length; i++) {
             claim = claims[i];
-            token = tokens[claim.tokenIndex];
 
-            require(!isClaimed(token, claim.distributor, claim.distribution, liquidityProvider), "cannot claim twice");
+            // When we process a new claim we either
+            // a) aggregate the new claim bit with previous claims of the same channel/claim bitmap
+            // b) set claim status and start aggregating a new set of currentBits for a new channel/word
+            if (currentChannelId == bytes32(0)) {
```

Comment:
I'd remove this if statement and change the else block to conditionally update the stored bitmap as so
```
if (channelId != bytes32(0)){
   _setClaimedBits(liquidityProvider, currentChannelId, currentWordIndex, currentBits);
   _deductClaimedBalance(currentChannelId, currentClaimAmount);
 }

This would remove us having to perform a check that's only relevant for the first claim on each iteration.